src/main/java/com/google/sps/data/Room.java

```diff
@@ -0,0 +1,123 @@
+    private List<Member> members;
+    private LinkedList<Message> messages;
+    private Queue<Video> videos;
+
+    /**
+      * Room constructor
+      * @param members a List of Member objects
+      * @param videos an Queue of video objects
+      * @return a new Room object
+      */
+    private Room(List<Member> members, Queue<Video> videos){
+        this.messages = new LinkedList<Message>();
+        this.members = members;
+        this.videos = videos;
+    }
+    /**
+      * Room constructor
+      * @param members a List of Member objects
+      * @param videos an Queue of video objects
+      * @param messages a List of message objects
+      * @return a new Room object
+      */
+    private Room(List<Member> members, Queue<Video> videos, LinkedList<Message> messages) {
+        this.messages = messages;
+        this.videos = videos;
+        this.messages = messages;
+    }
+    public static Room createRoom(List<Member> members, Queue<Video> videos, LinkedList<Message> messages){
+        return new Room(members, videos, messages);
+    }
+    //Returns the Room's members
+    public List<Member> getMembers() {
+        return this.members;
+    }
+
+    //Get all of the members as a list of EmbeddedEntities
+    private List<EmbeddedEntity> getMembersAsEmbeddedEntities(){
+        return members.stream().map(Member::toEmbeddedEntity).collect(Collectors.toList());
+    }
+
+    //Returns the Room's video url list
+    public Queue<Video> getVideos() {
+        return this.videos;
+    }
+
+    //Returns the Room's message list
+    public LinkedList<Message> getMessages(){
+        return this.messages;
+    }
+    //Returns the messages as a list of embedded entities
+    private LinkedList<EmbeddedEntity> getMessagesAsEntities() {
+        return (LinkedList) messages.stream().map(Message::toEmbeddedEntity).collect(Collectors.toList());
+    }
+    //Returns a queue of embedded entities
+    private Queue<EmbeddedEntity> getVideosAsEntities() {
+        return (Queue<EmbeddedEntity>) this.videos.stream().map(Video::toEmbeddedEntity).collect(Collectors.toList());
+    }
+    //Turns the Room object into a datastore entity
+    public static Entity toEntity(Room room){
+        Entity newRoom = new Entity(ROOM_ENTITY);
+        newRoom.setProperty(MEMBERS_PROPERTY, room.getMembersAsEmbeddedEntities());
+        newRoom.setProperty(VIDEOS_PROPERTY, room.getVideosAsEntities());
+        newRoom.setProperty(MESSAGES_PROPERTY, room.getMessagesAsEntities());
+        return newRoom;
+    }
+    public static Room fromKey(Key roomKey) {
+        try {
+        Entity roomEntity = datastore.get(roomKey);
+        return Room.fromEntity(roomEntity);
+        } catch (EntityNotFoundException e){
+            System.out.println(e.toString());
+        }
+        return null;
+    }
+    public void setMessages(LinkedList<Message> messages){
+        this.messages = messages;
+    }
```

Comment:
This function shouldn't be needed. External files shouldn't be able to set all the messages right? Only be able to add a new one as needed.